wandb/apis/public.py

```diff
@@ -2377,7 +2382,98 @@ def __repr__(self):
         return "<ArtifactCollection {} ({})>".format(self.name, self.type)
 
 
+def _determine_artifact_root(source_path, target_path):
+    """Helper function to determine the artifact root of `target_path` by comparing to
+    an existing artifact asset in `source`path`. This is used in reference artifact resolution"""
+    abs_source_path = os.path.abspath(source_path)
+    abs_target_path = os.path.abspath(target_path)
+
+    # Break the source path into parts
+    head = abs_source_path
+    source_path_parts = []
+    while head != "" and head != "/":
```

Comment:
This won't work on windows ("/"). Python has os.path.commonpath (only 3.5 or later) and os.path.commonprefix.

Reply:
Also here you'd just do os.path.split()

Reply:
fixed in next commit